src/chat.rs

```diff
@@ -1729,50 +1744,123 @@ pub async fn marknoticed_all_chats(context: &Context) -> Result<(), Error> {
     Ok(())
 }
 
+/// Emits an event to update the time until the next local deletion.
+///
+/// This takes into account only per-chat timeouts, because global device
+/// timeouts are at least one hour long and deletion is triggered often enough
+/// by user actions.
+pub async fn update_autodelete_timeout(context: &Context) {
+    let autodelete_timestamp: Option<i64> = match context
+        .sql
+        .query_get_value_result(
+            "SELECT autodelete_timestamp \
+         FROM msgs \
+         WHERE autodelete_timestamp != 0 \
+           AND chat_id != ? \
+         ORDER BY autodelete_timestamp ASC \
+         LIMIT 1",
+            paramsv![DC_CHAT_ID_TRASH],
+        )
+        .await
+    {
+        Err(err) => {
+            warn!(context, "Can't calculate next autodelete timeout: {}", err);
+            return;
+        }
+        Ok(autodelete_timestamp) => autodelete_timestamp,
+    };
+
+    let timer = if let Some(next_autodelete) = autodelete_timestamp {
+        let now = time();
+        if now > next_autodelete {
+            1
+        } else {
+            (next_autodelete - now)
+                .try_into()
+                .unwrap_or(u32::MAX)
+                .saturating_add(1)
```

Comment:
Both this, and because `next_autodelete` and `now` are integers, so if next autodelete happens at 5.7, and now is 4.4, we need to wait for 2 seconds, not 1.

Reply:
Will add a comment.